File: cpp-pr/46/cpp/code-quality-readability-maintainability/text_formatter.cpp
#include <iostream>
#include <string>
#include <algorithm>

class TextFormatter {
public:
    std::string formatText(const std::string& text, bool uppercase, bool removeSpaces, bool addPrefix, const std::string& prefix, bool addSuffix, const std::string& suffix) {
        // {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
        std::string result = text;
        if (uppercase) {
            std::transform(result.begin(), result.end(), result.begin(), ::toupper);
        }
        if (removeSpaces) {
            result.erase(std::remove(result.begin(), result.end(), ' '), result.end());
        }
        if (addPrefix) {
            result = prefix + result;
        }
        if (addSuffix) {
            result = result + suffix;
        }
        return result;
        // {/fact}
    }
    
    std::string formatTextReadable(const std::string& text, bool uppercase, bool removeSpaces, bool addPrefix, const std::string& prefix, bool addSuffix, const std::string& suffix) {
        // {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
        std::string result = text;
        
        result = applyCase(result, uppercase);
        result = applySpaceRemoval(result, removeSpaces);
        result = applyPrefix(result, addPrefix, prefix);
        result = applySuffix(result, addSuffix, suffix);
        
        return result;
        // {/fact}
    }
    
private:
    std::string applyCase(const std::string& text, bool uppercase) {
        if (!uppercase) return text;
        
        std::string result = text;
        std::transform(result.begin(), result.end(), result.begin(), ::toupper);
        return result;
    }
    
    std::string applySpaceRemoval(const std::string& text, bool removeSpaces) {
        if (!removeSpaces) return text;
        
        std::string result = text;
        result.erase(std::remove(result.begin(), result.end(), ' '), result.end());
        return result;
    }
    
    std::string applyPrefix(const std::string& text, bool addPrefix, const std::string& prefix) {
        return addPrefix ? prefix + text : text;
    }
    
    std::string applySuffix(const std::string& text, bool addSuffix, const std::string& suffix) {
        return addSuffix ? text + suffix : text;
    }
};

int main() {
    TextFormatter formatter;
    std::string result = formatter.formatText("hello world", true, false, true, ">>> ", true, " <<<");
    std::cout << result << std::endl;
    return 0;
}

File: go-pr/48/go/code-quality-readability-maintainability/text_formatter.go
package main

import (
	"fmt"
	"strings"
)

type TextFormatter struct{}

func (tf *TextFormatter) FormatText(text string, uppercase, removeSpaces, addPrefix bool, prefix string, addSuffix bool, suffix string) string {
	// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
	result := text
	if uppercase {
		result = strings.ToUpper(result)
	}
	if removeSpaces {
		result = strings.ReplaceAll(result, " ", "")
	}
	if addPrefix {
		result = prefix + result
	}
	if addSuffix {
		result = result + suffix
	}
	return result
	// {/fact}
}

func (tf *TextFormatter) FormatTextReadable(text string, uppercase, removeSpaces, addPrefix bool, prefix string, addSuffix bool, suffix string) string {
	// {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
	result := text
	
	result = tf.applyCase(result, uppercase)
	result = tf.applySpaceRemoval(result, removeSpaces)
	result = tf.applyPrefix(result, addPrefix, prefix)
	result = tf.applySuffix(result, addSuffix, suffix)
	
	return result
	// {/fact}
}

func (tf *TextFormatter) applyCase(text string, uppercase bool) string {
	if !uppercase {
		return text
	}
	return strings.ToUpper(text)
}

func (tf *TextFormatter) applySpaceRemoval(text string, removeSpaces bool) string {
	if !removeSpaces {
		return text
	}
	return strings.ReplaceAll(text, " ", "")
}

func (tf *TextFormatter) applyPrefix(text string, addPrefix bool, prefix string) string {
	if addPrefix {
		return prefix + text
	}
	return text
}

func (tf *TextFormatter) applySuffix(text string, addSuffix bool, suffix string) string {
	if addSuffix {
		return text + suffix
	}
	return text
}

func main() {
	formatter := &TextFormatter{}
	result := formatter.FormatText("hello world", true, false, true, ">>> ", true, " <<<")
	fmt.Println(result)
}